framework/include/userobject/LayeredBase.h

```diff
@@ -65,6 +65,17 @@ class LayeredBase : private Restartable
    */
   virtual unsigned int getLayer(Point p) const;
 
+  /**
+   * Get the center coordinates for the layers (along given direction)
+   */
+  const std::vector<Real> & getLayerCenters() const { return _layer_centers; }
+
+  /**
+   * Get direction of the layers
+   * @return layer direction
+   */
+  const unsigned int & direction() const { return _direction; }
```

Comment:
Does this need to be a reference? does it change?

Reply:
No, I just always do references for things that are member variables out of habit

Reply:
If you're returning something 8 bytes or less (the same size as a memory address), you should just return a copy instead of a reference. In this case, it subtly implies that `_direction` is changing because you're returning it as a ref